data-pipeline-ffi/src/trace_exporter.rs

```diff
@@ -0,0 +1,56 @@
+// Copyright 2024-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+use bytes::Bytes;
+use data_pipeline::trace_exporter::TraceExporter;
+use data_pipeline::trace_exporter::TraceExporterBuilder;
+use ddcommon_ffi::{
+    slice::{AsBytes, ByteSlice},
+    CharSlice,
+};
+use std::ffi::{c_char, CString};
+
+#[no_mangle]
+pub unsafe extern "C" fn dd_trace_exporter_new(
+    host: CharSlice,
+    port: u16,
+    tracer_version: CharSlice,
+    language: CharSlice,
+    language_version: CharSlice,
+    language_interpreter: CharSlice,
+) -> *mut TraceExporter {
+    let mut builder = TraceExporterBuilder::default();
+
+    let exporter = builder
+        .set_host(host.to_utf8_lossy().as_ref())
+        .set_port(port)
+        .set_tracer_version(tracer_version.to_utf8_lossy().as_ref())
+        .set_language(language.to_utf8_lossy().as_ref())
+        .set_language_version(language_version.to_utf8_lossy().as_ref())
+        .set_language_interpreter(language_interpreter.to_utf8_lossy().as_ref())
+        .build()
+        .unwrap();
+
+    Box::into_raw(Box::new(exporter))
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn dd_trace_exporter_free(ctx: *mut TraceExporter) {
+    if let Some(p) = ctx.as_mut() {
+        drop(Box::from_raw(p))
+    }
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn dd_trace_exporter_send(
+    ctx: *mut TraceExporter,
+    trace: ByteSlice,
+    trace_count: usize,
+) -> *const c_char {
+    let handle = Box::from_raw(ctx);
+    let response = handle
+        .send(Bytes::copy_from_slice(trace.as_bytes()), trace_count)
```

Comment:
So can't this use the bytes directly instead of doing a copy? Isn't there a way to create bytes from a `Box<[u8]>`, or should the `send` in `TraceExporter` really operate on `&[u8]` instead of `Bytes`?

Reply:
This should operate on &[u8], I mean if you use Bytes, it will also free it, which it must not.

Reply:
Good point. I didn't want to take ownership over the buffer just in case we needed to manipulate it. Maybe we can use COW semantics to use it and try to avoid the copy when it's not necessary. If you agree I will carry on with that change in another PR.